runtime/common/src/migrations/liquidity_pools_gateway.rs

```diff
@@ -41,77 +55,223 @@ pub type FailedOutboundMessages<T: pallet_liquidity_pools_gateway::Config + fram
+				);
+				weight.saturating_accrue(T::DbWeight::get().reads(1));
+			}
+
+			log::info!("{LOG_PREFIX}: Migration done!");
+
+			weight
 		}
 
-		for (nonce, entry) in FailedOutboundMessages::<T>::iter() {
-			log::warn!(
-				"{LOG_PREFIX}: Found failed outbound message:\nnonce:{nonce}\nentry:{entry:?}"
+		#[cfg(feature = "try-runtime")]
+		fn pre_upgrade() -> Result<Vec<u8>, sp_runtime::TryRuntimeError> {
+			// Extra check to confirm that the storage alias is correct.
+			assert_eq!(
+				OutboundMessageQueue::<T>::iter_keys().count(),
+				0,
+				"{LOG_PREFIX}: OutboundMessageQueue should be empty!"
 			);
-			weight.saturating_accrue(T::DbWeight::get().reads(1));
+
+			assert_eq!(
+				FailedOutboundMessages::<T>::iter_keys().count(),
+				0,
+				"{LOG_PREFIX}: FailedOutboundMessages should be empty!"
+			);
+
+			log::info!("{LOG_PREFIX}: Pre checks done!");
+
+			Ok(Vec::new())
 		}
 
-		log::info!("{LOG_PREFIX}: Migration done!");
+		#[cfg(feature = "try-runtime")]
+		fn post_upgrade(_: Vec<u8>) -> Result<(), sp_runtime::TryRuntimeError> {
+			assert_eq!(
+				OutboundMessageNonceStore::<T>::get(),
+				0,
+				"{LOG_PREFIX}: OutboundMessageNonceStore should be 0!"
+			);
 
-		weight
+			assert_eq!(
+				OutboundMessageQueue::<T>::iter_keys().count(),
+				0,
+				"{LOG_PREFIX}: OutboundMessageQueue should be empty!"
+			);
+
+			assert_eq!(
+				FailedOutboundMessages::<T>::iter_keys().count(),
+				0,
+				"{LOG_PREFIX}: FailedOutboundMessages should be empty!"
+			);
+
+			log::info!("{LOG_PREFIX}: Post checks done!");
+
+			Ok(())
+		}
 	}
+}
+
+pub mod clear_deprecated_domain_router_entries {
+	use frame_system::pallet_prelude::OriginFor;
+	use sp_arithmetic::traits::SaturatedConversion;
+
+	use super::*;
+	pub struct Migration<T>(PhantomData<T>);
+
+	const LOG_PREFIX: &str = "LPGatewayClearDomainRouters";
+
+	impl<T> OnRuntimeUpgrade for Migration<T>
+	where
+		T: pallet_liquidity_pools_gateway::Config
+			+ pallet_xcm_transactor::Config
+			+ pallet_ethereum_transaction::Config
+			+ pallet_evm::Config
+			+ frame_system::Config,
+		T::AccountId: AsRef<[u8; 32]>,
+		OriginFor<T>:
+			From<pallet_ethereum::Origin> + Into<Result<pallet_ethereum::Origin, OriginFor<T>>>,
+		v0::DomainRouter<T>: sp_std::fmt::Debug,
+		<T as pallet_liquidity_pools_gateway::Config>::Router:
+			From<liquidity_pools_gateway_routers::DomainRouter<T>>,
+	{
+		fn on_runtime_upgrade() -> Weight {
+			let items = v0::DomainRouters::<T>::iter_keys().count().saturated_into();
 
-	#[cfg(feature = "try-runtime")]
-	fn pre_upgrade() -> Result<Vec<u8>, sp_runtime::TryRuntimeError> {
-		// Extra check to confirm that the storage alias is correct.
-		assert_eq!(
-			OutboundMessageQueue::<T>::iter_keys().count(),
-			0,
-			"{LOG_PREFIX}: OutboundMessageQueue should be empty!"
-		);
+			pallet_liquidity_pools_gateway::DomainRouters::<T>::translate_values::<
+				v0::DomainRouter<T>,
```

Comment:
Could be this: `Option<v0::DomainRouter<T>>`? cc @wischli

Reply:
Unfortunately I had tried that already. Still unable to decode.